Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp

```diff
@@ -112,7 +113,6 @@ GstElementFactory* GStreamerVideoCapturer::getEncoder (const char* format)
     encoder_factory = GST_ELEMENT_FACTORY (g_list_first(encoders)->data);
```

Comment:
I'm really sorry for not having noticed this before, but you need an extra `gst_object_ref(encoder_factory)` here and `GRefPtr<GstElementFactory> h264encoder = adoptGRef(getEncoder ("video/x-h264"));` above in `createConverter()`. Otherwise, the object could reach zero references and be destroyed below, during the `gst_plugin_feature_list_free()` call. In that case, an invalid pointer would be returned by this method.

Even better than that, declare `GRefPtr<GstElementFactory> encoder_factory;` in this method, assign its value here as `encoder_factory = adoptGRef(GST_ELEMENT_FACTORY (g_list_first(encoders)->data));` and return encoder_factory leaving the return line unchanged in this method. You'll need to change the method signature to return a `GRefPtr<GstElementFactory>`. Leave the code in `createConverter()` unchanged, as the value will already be an adopted `GRefPtr`, no need to adopt it there anymore.

Reply:
Thank you for review.
The 2nd approach seemed direct, but gave me a run-time error:
"Trying to dispose object but it still has a parent.
You need to let the parent manage the object instead of unreffing the object directly."
So went ahead with 1st proposed solution.

Reply:
Oh, now I see it! The second solution `adoptGRef()` would own (steal) the reference, so the refcount could be set to zero in `gst_plugin_feature_list_free()`. The proper way to implement the second solution would have been as `encoder_factory = GST_ELEMENT_FACTORY (g_list_first(encoders)->data);`. That would increase the refcount by one, so at least one reference would exist after `gst_plugin_feature_list_free()`. That reference would be transferred to `createConverter()`.

Anyway, the first solution is fine, so the patch is good to merge.